File: Fourth Semester - Design and Programming Environments/C#/ProiectC#/model/validator/ReservationValidator.cs
namespace TravelAgency.model.validator;

using TravelAgency.exception;

public class ReservationValidator: IValidator<Reservation>
{
    public bool Validate(Reservation entity)
    {
        if (entity == null)
        {
            throw new ArgumentNullException(nameof(entity), "Reservation cannot be null");
        }

        if (entity.ClientName == null)
        {
            throw new ValidationException("Client name cannot be empty");
        }
        
        if (entity.ClientContact == null)
        {
            throw new ValidationException("Client contact cannot be empty");
        }
        
        if (entity.Trip == null)
        {
            throw new ValidationException("Trip cannot be empty");
        }
        
        if (entity.NumberOfReservedSeats == null)
        {
            throw new ValidationException("Number Of Reserved Seats cannot be empty");
        }
        return true;
    }
}

File: Fourth Semester - Design and Programming Environments/Java/ProiectJava/java/org/example/modul/validator/ReservationValidator.java
package org.example.modul.validator;


import org.example.exception.ValidationException;
import org.example.modul.Reservation;

public class ReservationValidator implements Validator<Reservation> {
    @Override
    public void validate(Reservation entity) {
        if (entity == null) {
            throw new IllegalArgumentException("Agency is null");
        }
        if (entity.getNumberOfReservedSeats() == null) {
            throw new ValidationException("Number of reserved places is null");
        }
        if (entity.getTrip() == null) {
            throw new ValidationException("Travel is null");
        }
    }
}
